editoast/src/client/stdcm_search_env_commands.rs

```diff
@@ -0,0 +1,444 @@
+use crate::modelsv2::electrical_profiles::ElectricalProfileSet;
+use crate::modelsv2::stdcm_search_environment::StdcmSearchEnvironment;
+use crate::modelsv2::timetable::Timetable;
+use crate::modelsv2::work_schedules::WorkScheduleGroup;
+use crate::modelsv2::Infra;
+use crate::modelsv2::Scenario;
+use crate::CliError;
+use crate::Exists;
+use crate::Model;
+use crate::Retrieve;
+use chrono::Duration;
+use chrono::NaiveDateTime;
+use clap::Args;
+use clap::Subcommand;
+use editoast_models::DbConnection;
+use editoast_models::DbConnectionPoolV2;
+use std::error::Error;
+
+#[derive(Subcommand, Debug)]
+pub enum StdcmSearchEnvCommands {
+    SetFromScenario(SetSTDCMSearchEnvFromScenarioArgs),
+    SetFromScratch(SetSTDCMSearchEnvFromScratchArgs),
+    Show,
+}
+
+pub async fn handle_stdcm_search_env_command(
+    command: StdcmSearchEnvCommands,
+    db_pool: DbConnectionPoolV2,
+) -> Result<(), Box<dyn Error + Send + Sync>> {
+    let conn = &mut db_pool.get().await?;
+    match command {
+        StdcmSearchEnvCommands::SetFromScenario(args) => {
+            set_stdcm_search_env_from_scenario(args, conn).await
+        }
+        StdcmSearchEnvCommands::SetFromScratch(args) => {
+            set_stdcm_search_env_from_scratch(args, conn).await
+        }
+        StdcmSearchEnvCommands::Show => show_stdcm_search_env(conn).await,
+    }
+}
+
+macro_rules! exists_or_return {
```

Comment:
Instead of a macro, could we use a simple function taking an `Exists<K>` from `src/modelsv2/prelude/retrieve.rs`?

Reply:
Why do you want to use a function instead ?
Having `check_existence(WorkScheduleGroup, work_schedule_group_id, conn)?` seems as readable as `exists_or_return!(WorkScheduleGroup, work_schedule_group_id, conn)` to me

If I'm not mistaken, with a function I wouldn't benefit from `stringify!`, right ?
I also feel like a function returning a Result made only for being `?`ed would have a purpose a bit less clear than the macro

WDYT ?

Reply:
Okay, while re-reading, it does seem a bit more readable since it seems more rustic.

However I feel like the last two points still outweigh the readability improvement

Reply:
I dislike using metaprogramming unless it's actually required. The macro itself is less readable than a regular function, it requires juggling with more complicated constructs, error messages are tricky to decipher, it slows down the build, etc etc.

Reply:
Could you please provide me an example on how to retrieve the class's name without the macro ? I found [`type_name`](https://doc.rust-lang.org/std/any/fn.type_name.html) but fully qualified struct names aren't desirable.

I don't see how to have a meaningful error message inside the function, so I would guess I could  just put the code from the macro everywhere it was used, but that's an awful lot of boilerplate

Reply:
You could always split at `:` with `type_name`.

```rust
std::any::type_name::<Option<bool>>().rsplit_once(':').unwrap().1
```

Reply:
Isn't it preferable to have boilerplate rather than a potential panic ?

Reply:
I'll add that we do use technical `struct` names for user facing content: it's probably wrong. Instead of `❌ ElectricalProfileSet not found, id: 42`, a user might want a sentence with English words, so something like `❌ electrical profile not found, id '42'`.

Don't get me wrong, I'm fine with displaying the type like you propose (we don't take the necessary time to consider a real user exposed to these messages anyway). But with that argument in mind, I'm not sure that `ElectricalProfileSet` is objectively much better than `editoast::modelsv2::electrical_profiles::ElectricalProfileSet` both are technical anyway (although one is arguably shorter so maybe more aesthetic and less noisy, but that starts to be more subjective).

Reply:
If we want to display a human-readable name for the error message, we could pass it as an argument of the function?

Reply:
Fine